extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java

```diff
@@ -134,9 +136,12 @@ private void calculate() {
                 port = -1;
             }
 
-            String hostHeader = delegate.getHeader(X_FORWARDED_HOST);
-            if (hostHeader != null) {
-                setHostAndPort(hostHeader.split(",")[0], port);
+            if (proxyConfig.enableForwardedForHostHeader) {
+                AsciiString forwardedHeader = AsciiString.cached(proxyConfig.forwardedHostHeader);
```

Comment:
It would be better to do this in the recorder, and pass it into the parser, so it is only done once and not once per request.

Reply:
done